app/builders/contact_inbox_builder.rb

```diff
@@ -59,11 +59,13 @@ def twilio_source_id
   end
 
   def create_contact_inbox
-    ::ContactInbox.create_with(hmac_verified: hmac_verified || false).find_or_create_by!(
+    attrs = {
       contact_id: @contact.id,
       inbox_id: @inbox.id,
       source_id: @source_id
-    )
+    }
+
+    ::ContactInbox.where(attrs).first_or_create!(hmac_verified: hmac_verified || false)
```

Comment:
find_or_create_by(!) was rewritten in 7.1 to call create_or_find_by(!) internally.  hence this broke. 

ref: https://github.com/rails/rails/issues/51052